### PR TITLE
Result api

### PR DESCRIPTION
**Pull request by avodovnik**
### Issue: #493 

- [x ] Tick here to confirm that you have run RunCodeFormatter.ps1

### Description:
The pull request is a bit "loose", as agreed with @krist00fer. It just contains the skeleton, so that potentially work can continue. I've added an IResultsReader, that, for now, has one method (GetLatest) which returns an enumerable of Messages. I also refactored the OutputFormatter and renamed it to MessageFormatter and added the Parse call, which allows us to reuse the same code across input/output operations (i.e. use CSV and Json Formatters when reading the results).

### Test:
At the moment, there is nothing concrete to test, as it's only interface level. However, due to refactoring, it's worth making sure existing output managers work as intended.

### CODE REVIEW

**Comment by krist00fer** [comment]
I'm not sure why the build failed. I'll have a look at it.

**Comment by krist00fer** [comment]
Looks good so far, lets's se how it all turns out

**Comment by avodovnik** [comment]
I implemented an example for the reader, however I hardcoded the file-path in this push.

**Comment by avodovnik** [comment]
@krist00fer the latest push should have all changes expected in this, for the initial attempt (only file reader though, for now). I had to make quite a lot of changes across several interfaces, but nothing should be broken because of it as it is simply splitting some parts of the functionality, like in the File Algorithm, enabling me to get paths out, even if there is no message, etc. In any case, please review again, and let me know if you have any additional changes you think I should make, or if we should go ahead and merge this?

**Comment by krist00fer** [comment]
Just wanted to ping you and see if I could be of services with this. Let me know if you need help.

**Comment by avodovnik** [comment]
@krist00fer thanks for the ping - I finally pushed the changes, and updated the algorithm to work with partitioning - hopefully. At the moment, I think it might be more of a WIP than I originally thought. There was one assumption here that might be incorrect: partition ordering is irrelevant. If this isn't the case, I need to make a modification of `FileResultsReader` (line 79) to not yield the messages, but collate them, re-order them, and then yield them. The problem for that, though, is that it will require loading **all** file's messages in memory, to parse them, which I'd rather avoid if possible.

File: src/TestClients/AnalyticsTestClient/Menus/ResultsApiConsumerMenu.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                var clusteringSerializer = new CsvMessageFormatter("id", "type", "version", "enqueueTimeUtc", "gameSession", "lat", "lon", "geoHash", "geoHashPrecision", "geoHashCenterLat", "geoHashCenterLon", "rnd");
+
+
+                MenuItems.Add('1', new ConsoleMenuItem("DAU", () => GetLatestFromFileSystem(dauSerializer, "dau", "session-start")));
+                MenuItems.Add('2', new ConsoleMenuItem("Clustering", () => GetLatestFromFileSystem(clusteringSerializer, "clustering", "geo-location")));
+            }
+
+            public void GetLatestFromFileSystem(IMessageFormatter formatter, string pipeline, string messageType)
+            {
+                IFilePathAlgorithm _filePathAlgorithm = new PipelineDateFilePathAlgorithm(newFileOption: NewFileNameOptions.Every5Minutes);
```
Comment (krist00fer) [inline src/TestClients/AnalyticsTestClient/Menus/ResultsApiConsumerMenu.cs]: Don't use underscore as a prefix for naming local variables (only for private class fields)

**Pull request merged**